### PR TITLE
BACKLOG-12927 : cleanup SearchControlBar and write small test to veri…

### PR DESCRIPTION
**Pull request by Yousria**
…fy multiple selection actions displayed

## JIRA

https://jira.jahia.org/browse/BACKLOG-12927

## Description
cleaned up the SearchControlBar which is part of the Toolbar and it was impossible to write a Jest test with the old version (wrapped with withStyles, connect and so on)
the test is only checking display of selection actions 

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+jest.mock('connected-react-router', () => jest.fn(() => {}));
+
+describe('Toolbar', () => {
+    it('should not render multiple selection actions', () => {
+        useSelector.mockImplementation(() => ({
+            selection: []
+        }));
+
+        const toolbar = shallow(<ToolBar/>);
+        expect(toolbar.find('Typography').length).toEqual(0);
```
Comment (cedricmoitrier) [inline src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: Could with have dedicated tests for the "selection-infos" instead? It's not clear what all those Typography related tests are about while just looking at the tests here.

You could have one test that check that the message is not displayed when there is not selection, and another one testing that it is displayed if there is any selection.

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        expect(toolbar.find('Button').length).toEqual(0);
+    });
+
+    it('should render multiple selection actions for 1 item', () => {
+        useSelector.mockImplementation(() => ({
+            selection: ['/test']
+        }));
+
+        const toolbar = shallow(<ToolBar/>);
+        expect(toolbar.find('Typography').prop('children')).toContain('itemsSelected');
```
Comment (cedricmoitrier) [inline src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: I would mock the "translate" function  and check it has been called with the expected parameters (label key and number of items) instead.

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        expect(toolbar.find('Button').prop('icon').type('SvgCancel')).toBeDefined();
+        expect(toolbar.find('Typography').prop('data-cm-selection-size')).toBe(1);
+        expect(toolbar.find('ButtonGroup').prop('children').length).toBe(4);
```
Comment (cedricmoitrier) [inline src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: You could use expect(...).toHaveLength(4) to test the length of an array

Comment (cedricmoitrier) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: Actually, I think you could even just write:
expect(toolbar.find('ButtonGroup').children()).toHaveLength(4);

Comment (Yousria) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: ok will use that

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx (hunk 2 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: ok, will see how to do that

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx (hunk 1 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: sure

File: src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx (hunk 2 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: couldn't mock it, as soon as I try to mock react-i18next, it's breaks all my tests, due to some buttons coming from the old design system

Comment (cedricmoitrier) [reply src/javascript/JContent/ContentRoute/ToolBar/Toolbar.test.jsx]: Ok, fine. It was just a suggestion.

**Pull request merged**